en/cpp/guide/build.md

```diff
@@ -68,16 +68,58 @@ Make sure to get all the submodules as well:
 git submodule update --init --recursive
 ```
 
+### Building Fundamentals
+
+Building the MAVSDK with CMake is broken down into two steps: **Configuration** and **Build**.
+
+#### Build Configuration Step {#configuration_step}
+For configuration, you specify the type of build you want to execute in the [build step](#build_step). You can checkout the [CMake Documentation on Build Configuration](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#build-configurations) for more context.
+
+A typical configuration command example would be:
+
+`cmake -DCMAKE_BUILD_TYPE=Debug -Bbuild/default -H.`
+
+- Build type is set to `Debug`
+- Build directory is set to `build/default`
+- TODO: Explain `H.` flag
+
+During the configure step you can set more flags using `-DFLAG=Value`:
+
+- `CMAKE_BUILD_TYPE`: as documented above, to chose between `Debug` and `Release` build.
+- `CMAKE_INSTALL_PREFIX`: as documented above, to specify directory to install library artifacts.
+- `BUILD_SHARED_LIBS`: set to `ON` to build dynamic libraries (such as .so on Linux, .dylib on macOS, .dll on Windows). Set to `OFF` to build static libraries (such as .a on Linux and macOS, .lib on Windows).
+- `SUPERBUILD`: set to `OFF` to use system dependencies instead of [third party dependencies](https://github.com/mavlink/MAVSDK/tree/main/third_party) downloaded and built using cmake.
+- `CMAKE_PREFIX_PATH`: can be used to set the path where the dependencies can be found if `SUPERBUILD` is set to `OFF`.
+- `BUILD_MAVSDK_SERVER`: set to `ON` to build mavsdk_server, see instruction to [build mavsdk_server](build_mavsdk_server.md).
+- `ASAN`: set to `ON` to enable address sanitizer.
+- `UBSAN`: set to `ON` to enable undefined behavior sanitizer.
+- `LSAN`: set to `ON` to enable leak sanitizer.
+- `WERROR`: set to `ON` to error on warnings, mostly used for CI.
+
+After the configuration step, everything that will be build in the [build step](#build_step) have been specified, and if you want to change your build configuration (e.g. If you want to build `Debug` build instead of `Release` build), you must execute the [configuration step](#configuration_step) again!
```

Comment:
Here, I was conflicted since below, it says "If you already have run *cmake* without setting `CMAKE_INSTALL_PREFIX`, you may need to clean the build first:".

Which implies that doing the configuration step again isn't enough when you want to change the CMAKE_INSTALL_PREFIX setting.

I think that the 'you may need to clean the build first' part should be removed (it is wrong?!), if my understanding that `rm -rf build/default` is only necessary when the `third party` changes are present. Thoughts on this?

Reply:
I think it works to just re-run the configure step (no need to clean build) :+1:. Not 100% sure it always does, though.